test/inductor/test_compiled_optimizers.py

```diff
@@ -343,66 +342,75 @@ def test_correctness(self, device, dtype, optim_info):
         all_optim_inputs = _get_optim_inputs_including_global_cliquey_kwargs(
             device, dtype, optim_info, skip=("differentiable",)
         )
-        for optim_input in all_optim_inputs:
-            kwargs = optim_input.kwargs
-
-            torch._dynamo.reset()
-            torch._inductor.metrics.reset()
-            input = torch.ones([10, 10], device=device)
-            model_eager = torch.nn.Sequential(
-                *[torch.nn.Linear(10, 10, device=device) for _ in range(2)]
-            )
-            model_eager(input).sum().backward()
-            model_compiled = deepcopy(model_eager)
-            model_compiled(input).sum().backward()
-
-            if optim_cls is SparseAdam:
-                for param in model_eager.parameters():
-                    param.grad = param.grad.to_sparse()
-                for param in model_compiled.parameters():
-                    param.grad = param.grad.to_sparse()
-
-            opt_compiled = optim_cls(model_compiled.parameters(), **kwargs)
-            opt_eager = optim_cls(model_eager.parameters(), **kwargs)
-
-            if optim_cls is LBFGS:
-
-                @torch.compile()
-                def fn():
-                    def closure():
-                        loss = model_compiled(input).sum()
+        closure_options = (True,) if optim_info.step_requires_closure else (False, True)
```

Comment:
Better to use parametrization over the test similar to with_lrsched here: https://github.com/pytorch/pytorch/blob/main/test/test_optim.py#L82

It is easier to read and will make it separate tests while allowing you to reuse the code

Reply:
I would like to skip the parameterization for the requires closure optimizers though. Don't see a way to do that with parameterizations..